source/develop/release-management/features/virt/enhance-import-export-with-ova.md

```diff
@@ -0,0 +1,139 @@
+2. The front-end sends the OVF to the back-end.
+3. The back-end parses the OVF and returns back an object that represents the VM.
+4. The front-end displays the VM configuration and allows the user to configure the parameters for the import (namely, VM name, destination storage, and networks).
+
+From this point on, the process depends on the source of the OVA. Note that the source is returned to the front-end by the back-end in the _origin_ field of the VM.  
+
+### Upload OVA generated by oVirt
+If the OVA was generated by oVirt, the rest of the process will be:
+
+1. The back-end adds the VM entity into the database.
+2. The back-end creates the destination volumes for the VM disks using the existing process for this in oVirt.
+3. The back-end requests the imageio proxy to prepare to upload-ova with a list of parameters per disk. Namely, these parameters contain the destination path of the disk and its actual size.
+4. The proxy returns a URL (similar to the upload-image flow).
+5. The back-end passes this URL to the front-end (similar to the upload-image flow).
+6. The front-end starts transmitting the **disks** one by one.
+7. The proxy receives the input stream and streams it to the relevant destination volume.
+
+Note that generating OVA files with compressed disks are out of the scope of this feature. In order to support this, we will need to pass a flag to the proxy that indicates whether or not the disk is compressed (step 3) so the proxy could uncompress the data on-the-fly while streaming it to the destination volume (step 7).
+
+### Upload OVA generated by an external source
+If the OVA was generated by external source, the rest of the process will be:  
+
+1. The back-end adds the VM entity into the database.
+2. The back-end creates a temporary volume to store the OVA in.
+3. The back-end requests (through VDSM) the imagoio proxy to prepare to upload-image-like process.
+4. The proxy returns a URL (similar to the upload-image flow).
+5. The back-end passes this URL to the front-end (similar to the upload-image flow).
+6. The front-end starts transmitting the **whole OVA**.
+7. The proxy streams the input into the created volume.
+8. When the streaming is done, the back-end triggers the existing convert OVA method.
+9. The back-end asks to remove the temporary volume upon completion.
+
+Note that this is a second-priority task.  
+
+## Export as OVA
+The user will be able to specify a VM or a template to be exported as an OVA. In that case, the user needs to provide a path on a host where the OVA file is going to be created. The export process will then be comprised of the following operations:  
+
+1. The front-end passes the back-end a VM (or template) to export and a path on a host that the OVA should be created in.
+2. Each of the disks of the VM/template that either contains more than one volume or whose volume is not raw will be collapsed and converted to raw at some storage domain (TBD: sparsed?).
```

Comment:
1. - right, we should avoid additional conversions. That's my point - if the source is qcow why convert it to raw? Especially when qcow supports internal compression
2. - yes, in expense of being far larger, which is quite a drawback when you move data around
3. - I would not presume what are 3rd party tools consuming. Outside of KVM world qcow is indeed not so common, but in VMware works it's mostly their native vmdk anyway... But yes, for greatest compatibility raw would be good (even though it may mean conversion and larger disk space), but for any other use case (archiving,ovirt<->ovirt) it is bad